plugins/modules/host.py

```diff
@@ -0,0 +1,341 @@
+            returned: optional
+        host_id:
+            description: A unique host identifier. This is not the same as the hostname (FQDN). It is a distinct value that remains the same even if the hostname changes.
+            type: str
+            returned: optional
+        host_url:
+            description: A URL into the Cloudera Manager web UI for this specific host.
+            type: str
+            returned: optional
+        hostname:
+            description: The hostname. This field is not mutable after the initial creation.
+            type: str
+            returned: optional
+        ip_address:
+            description: The host IP address. This field is not mutable after the initial creation.
+            type: str
+            returned: optional
+        last_heartbeat:
+            description: Time when the host agent sent the last heartbeat.
+            type: str
+            returned: optional
+        maintenance_mode:
+            description: Maintance mode of Cloudera Manager Service.
+            type: bool
+            returned: optional
+        maintenance_owners:
+            description: List of Maintance owners for Cloudera Manager Service.
+            type: list
+            returned: optional
+        num_cores:
+            description: The number of logical CPU cores on this host.
+            type: number
+            returned: optional
+        numPhysicalCores:
+            description: The number of physical CPU cores on this host.
+            type: number
+            returned: optional
+        rack_id:
+            description: The rack ID for this host.
+            type: str
+            returned: optional
+        role_refs:
+            description: The list of roles assigned to this host.
+            type: list
+            returned: optional
+        tags:
+            description: Tags associated with the host.
+            type: list
+            returned: optional
+        total_phys_mem_bytes:
+            description: he amount of physical RAM on this host, in bytes.
+            type: str
+            returned: optional
+"""
+
+
+class ClouderaHost(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaHost, self).__init__(module)
+
+        # Initialize the return values
+        self.host_name = self.get_param("host_name")
+        self.cluster_name = self.get_param("cluster_name")
+        self.host_ip = self.get_param("host_ip")
+        self.state = self.get_param("state")
+        self.rack_id = self.get_param("rack_id")
+        # Execute the logic
+        self.process()
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+
+        cluster_api_instance = ClustersResourceApi(self.api_client)
+        host_api_instance = HostsResourceApi(self.api_client)
+        self.host_output = {}
+        self.changed = False
+        existing = None
+
+        try:
+            hosts = host_api_instance.read_hosts().to_dict()
```

Comment:
TIL that the `read_host()` API call accepts either `host_id` or `hostname`, which would save us the pain of iterating over a potentially giant host list.